server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/ContactExtractorTest.java

```diff
@@ -0,0 +1,166 @@
+                        .build())
+                    .addMailet(MailetConfiguration.builder()
+                        .matcher(All.class)
+                        .mailet(RemoveMimeHeader.class)
+                        .addProperty("name", "bcc")
+                        .build())
+                    .addMailet(MailetConfiguration.builder()
+                        .matcher(RecipientIsLocal.class)
+                        .mailet(VacationMailet.class)
+                        .build())
+                    .addMailet(MailetConfiguration.builder()
+                        .matcher(RecipientIsLocal.class)
+                        .mailet(LocalDelivery.class)
+                        .build())
+                    .build())
+            .build();
+        jamesServer = new TemporaryJamesServer(folder, mailets);
+        DataProbeImpl probe = jamesServer.getProbe(DataProbeImpl.class);
+        probe.addDomain(JAMES_ORG);
+        probe.addUser(SENDER, PASSWORD);
+        probe.addUser(TO, PASSWORD);
+        probe.addUser(TO2, PASSWORD);
+        probe.addUser(CC, PASSWORD);
+        probe.addUser(CC2, PASSWORD);
+        probe.addUser(BCC, PASSWORD);
+        probe.addUser(BCC2, PASSWORD);
+    }
+
+    @After
+    public void tearDown() {
+        jamesServer.shutdown();
+    }
+
+    @Test
+    public void recipientsShouldBePublishedToAmqpWhenSendingEmail() throws Exception {
+        MimeMessage message = MimeMessageBuilder.mimeMessageBuilder()
+            .setSender(SENDER)
+            .addToRecipient(TO, "John To2 <" + TO2 + ">")
+            .addCcRecipient(CC, "John Cc2 <" + CC2 + ">")
+            .addBccRecipient(BCC, "John Bcc2 <" + BCC2 + ">")
+            .setSubject("Contact collection Rocks")
+            .setText("This is my email")
+            .build();
+        FakeMail mail = FakeMail.builder()
+            .mimeMessage(message)
+            .sender(new MailAddress(SENDER))
+            .recipients(new MailAddress(TO), new MailAddress(TO2), new MailAddress(CC), new MailAddress(CC2), new MailAddress(BCC), new MailAddress(BCC2))
+            .build();
+        try (SMTPMessageSender messageSender = SMTPMessageSender.authentication("localhost", 1025, JAMES_ORG, SENDER, PASSWORD);
+                IMAPMessageReader imap = new IMAPMessageReader("localhost", 1143)) {
+
+            messageSender.sendMessage(mail);
+            Awaitility.await().pollDelay(Duration.FIVE_HUNDRED_MILLISECONDS)
+                .atMost(Duration.ONE_MINUTE)
+                .until(() -> imap.userReceivedMessage(TO, PASSWORD));
+
+            Optional<String> actual = amqpRule.readContent();
+            assertThat(actual).isNotEmpty();
+            assertThatJson(actual.get()).isEqualTo("{"
+                    + "\"userEmail\" : \"sender@james.org\", "
+                    + "\"emails\" : [ \"to@james.org\", \"John To2 <to2@james.org>\", \"cc@james.org\", \"John Cc2 <cc2@james.org>\", \"bcc@james.org\", \"John Bcc2 <bcc2@james.org>\" ]"
```

Comment:
Are you sure it is well handled by OpenPaaS?

Reply:
Should we handle line wrapping and character escaping?

Reply:
We don't really know for those points, let see.

Reply:
Couldn't we rather ask Chamerling?

Reply:
Checked, it is supported by the client.
http://docs.open-paas.org/modules/contact/collected/#message-with-useremail